src/main/java/dk/sdu/se_f22/productmodule/management/ProductManager.java

```diff
@@ -0,0 +1,356 @@
+
+import dk.sdu.se_f22.sharedlibrary.models.Product;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class ProductManager implements IProductManager, Runnable{
+
+    private final JSONReader jsonReader;
+
+    public ArrayList<Product> productArray;
+    private ArrayList<Product> updatedProductArray;
+    private boolean backgroundThreadIsRunning = false;
+    private final Thread backgroundThread;
+    private long lastCall;
+
+    private int updateInterval;    //Measured in minutes
+    private final String config = "src/main/resources/dk/sdu/se_f22/productmodule/management/config.txt";
+
+    public ProductManager(String sourcePath){
+
+        //When you create a new object of this class, the background thread is started automatically.
+        //If you want to alter an attribute on a product, you MUST do this through the C.U.D. functions in this class.
+
+        jsonReader = new JSONReader(sourcePath);
+        productArray = getFromSource();
+        updatedProductArray = null;
+
+        backgroundThread = new Thread(this);
+        backgroundThread.start();
+
+        lastCall = System.currentTimeMillis();
+        updateInterval = readConfig();
+    }
+    
+    ///Default File Path
+    public ProductManager(){
+        this("src/main/resources/dk/sdu/se_f22/productmodule/management/products.json");
+    }
+
+    public static void main(String[] args) {
+
+        //This is only used for testing right now.
+
+        ProductManager manager = new ProductManager();
+        try {
+            manager.productArray = manager.jsonReader.read();
+            manager.jsonReader.write(manager.productArray);
+        }catch (IOException e){
+            e.printStackTrace();
+        }
+
+
+        manager.printAllProducts();
+        // write your code here
+    }
+
+    @Override
+    public boolean create(Product p) {
+
+        //Adds new product to the productArray.
+        //Returns whether this was possible or not.
+
+        checkForUpdates();
+        boolean success = productArray.add(p);
+        updateSource();
+
+        return success;
+    }
+
+    @Override
+    public boolean createAll(ArrayList<Product> pList){
+        checkForUpdates();
+
+        boolean success = productArray.addAll(pList);
+
+        updateSource();
+        return success;
+    }
+
+    @Override
+    public Product read(String productId) {
+
+        //This function returns a single product based on the UUID
+        //Since the read() function doesn't alter any attribute values on the product
+        //There's no reason to update the source.
+
+        checkForUpdates();
+
+        Product toReturn = null;
+
+        for(Product p : productArray){
+            if(p.get(ProductAttribute.UUID).equalsIgnoreCase(productId)){
+                toReturn = p;
+                break;
+            }
+        }
+
+        return toReturn;
+    }
+
+    @Override
+    public Product[] readAll(String[] productIds) {
```

Comment:
If it's based on the product UUIDS then make the method name more descriptive, so it comes off as it's only going to read the products you specify, furthermore make the array type of `UUID`.